File: C++/FastSolution.cpp
#include <iostream>

//a very fast solution that has a time complexity of O(sqrt(n))
bool primalityTest(int num) {
    if (!(num%2)) return num==2;

    for (int i = 3; i*i<=num; i+=2) {
        if (!(num%i)) return false;
    }    
    return true;
}


int main() {
    int num;
    std::cout << "Enter a number to check if it's prime or not.\n";
    std::cin >> num;
    std::cout << "The number you entered is "<<(primalityTest(num) ? "prime" : "not prime")<< "\n";
    return 0;   
}

File: Javascript/FastSolution.js
//relatively fast solution, O(n^(1/2)) time notation
function primalityTest(num) {
    if (num <= 1) return false;
    if (num%2==0) return num==2;
    for (let i = 3; i*i<=num; i+=2) {
        if (num%i==0) return false;
    }
    return true;
}

const test = 10; // number to check
console.log("The number "+test+" is "+(primalityTest(test)?"prime":"not prime"));
